### PR TITLE
added balance to analytics

### LINKED ISSUE #8057: Amplitude should record the user's balance (amount of gems)
Our  Amplitude analytics data should include the user's number of gems (the user's `balance` object) to help us track payment / purchase problems.

Ideally it should report it as something like this `Balance   0.75 (3 Gems)` to avoid us making mistakes with the mental arithmetic - I know it's not hard, but it's even easier for the computer. ;)

If for any reason the balance doesn't exist or isn't a finite number, the code should report that (I have no idea if that's possible with our current code base or any future code, but it wouldn't be good to report a balance of 0 if there was actually some bug that prevented the balance existing at all).

**EDIT 2016-12-27: The work remaining to do is:**

- Leave the existing "balance" property as it is now.
- Create a new property called "balance gem amount" or similar (spaces are optional) which works in the same way as the "balance" property but displays the balance divided by 4. I want the word "balance" (lower case) to be at the start of the property name so that in Amplitude it appears immediately under the existing "balance" field (sorting is alphabetical there).

### PR DESCRIPTION
**Pull request by dumindux**
Fixes #8057
### Changes

---

UUID:

### CODE REVIEW

File: website/server/libs/analyticsService.js [hunk 1]
```diff
@@ -66,6 +66,10 @@ let _formatUserData = (user) => {
     properties.Mana = Math.floor(user.stats.mp);
   }
 
+  if (user.balance) {
```
Comment (paglias) [inline website/server/libs/analyticsService.js]: User.balance should Always exist. Also this check will Always returns false for a balsnce of 0 but we want to store it even if 0

**Review by paglias — CHANGES_REQUESTED** [review]

File: website/server/libs/analyticsService.js (hunk 1 above)
Comment (crookedneighbor) [reply website/server/libs/analyticsService.js]: Agreed.

Comment (dumindux) [reply website/server/libs/analyticsService.js]: Thanks for the suggestions. I will add the changes

**Comment by crookedneighbor** [comment]
Don't forget to provide your user id.

You can find it in [Settings > API](https://habitrpg.com/#/options/settings/api). Be sure to only copy the User ID and _not_ your API Token.

**Review by paglias — APPROVED** [review]

**Comment by paglias** [comment]
LGTM 👍

**Comment by crookedneighbor** [comment]
Let me know your user id so I can credit you for your work!

**Comment by dumindux** [comment]
Hi, my user id is 5fc49b3a-752c-4d35-9f73-287e19fdc96b

**Comment by crookedneighbor** [comment]
Tier awarded! You can read more about [contributor tiers and rewards here](http://habitrpg.wikia.com/wiki/Contributor_Rewards).

Note that as your tier increases, it takes progressively more work to achieve the next one, so for the higher tiers, larger PRs or a few PRs are required. There's some information about that at Contributor Rewards, but feel free to ask here if you have questions!

**Comment by dumindux** [comment]
Wow great!!. Thank you :).

**Pull request merged**